apps/federatedfilesharing/lib/DiscoveryManager.php

```diff
@@ -141,22 +141,30 @@ private function ocmDiscover($remote) {
 		$decodedService = $this->makeRequest($remote . '/ocm-provider/');
 		if (!empty($decodedService) && $decodedService['enabled'] === true) {
 			$discoveredServices['ocm'] = $decodedService['endPoint'];
-			$shareTypes = $decodedService['shareTypes'];
-			foreach ($shareTypes as $type) {
-				if ($type['name'] == 'file') {
-					$discoveredServices['webdav'] = $type['protocols']['webdav'];
+			if (isset($decodedService['shareTypes'])) {
+				$shareTypes = $decodedService['shareTypes'];
+				foreach ($shareTypes as $type) {
+					if ($type['name'] == 'file') {
+						$discoveredServices['webdav'] = $type['protocols']['webdav'];
+					}
+				}
+			} elseif (isset($decodedService['resourceTypes'])) {
+				$resourceTypes = $decodedService['resourceTypes'];
```

Comment:
```suggestion
				$shareTypes = $decodedService['resourceTypes'];
```
And then the 2 copies of `foreach` can become a single loop after the if/elseif/else. (To protect, it would need `$shareTypes = []` in an `else` so that the loop always has some iterable in `$shareTypes`

Reply:
@phil-davis changed with sth similar. Thanks.